include/itkWaveletCoeffsPhaseAnalyzisImageFilter.hxx

```diff
@@ -0,0 +1,164 @@
+  // m_ForwardFFTFilter = FFTForwardType::New();
+  // m_ForwardWaveletFilter = ForwardWaveletType::New();
+
+  // m_MonogenicSignalFrequencyFilter = MonogenicSignalFrequencyType::New();
+  // m_VectorInverseFFTFilter = VectorInverseFFTType::New();
+  // m_PhaseAnalysisFilter = PhaseAnalysisType::New();
+  // m_FFTForwardPhaseFilter = FFTForwardType::New();
+
+  // m_InverseWaveletFilter = InverseWaveletType::New();
+  // m_InverseFFTFilter = InverseFFTType::New();
+
+  // m_ChangeInformationFilter = ChangeInformationType::New();
+  // m_CastFloatFilter = CastFloatType::New();
+
+  // ====================================================================
+  // ==================== Graft Input Declaration =======================
+  // ====================================================================
+  typename ImageType::Pointer input = ImageType::New();
+  input->Graft(const_cast<ImageType*> ( this->GetInput() ));
+  m_FFTPadFilter->SetInput( input );
+
+  // ==================== Filter Inter Connection =======================
+  m_ZeroDCFilter->SetInput( m_FFTPadFilter->GetOutput() );
+  m_ForwardFFTFilter->SetInput( m_ZeroDCFilter->GetOutput() );
+  m_ForwardWaveletFilter->SetInput( m_ForwardFFTFilter->GetOutput() );
+
+  // ==================== Filter Set Parameters =========================
+  m_ForwardWaveletFilter->SetHighPassSubBands( this->m_HighPassSubBands );
+  m_ForwardWaveletFilter->SetLevels( this->m_Levels );
+
+  // ======================== Loop the Filters ==========================
+  typename ForwardWaveletType::OutputsType        modifiedWavelets;
+  for ( unsigned int i = 0; i < m_ForwardWaveletFilter->GetNumberOfOutputs(); ++i )
+    {
+    m_MonogenicSignalFrequencyFilter->SetInput( m_ForwardWaveletFilter->GetOutput(i) ); // This API is strange.
+    m_VectorInverseFFTFilter->SetInput( m_MonogenicSignalFrequencyFilter->GetOutput() );
+    m_PhaseAnalysisFilter->SetInput( m_VectorInverseFFTFilter->GetOutput() );
+
+    m_PhaseAnalysisFilter->SetApplySoftThreshold( this->m_ApplySoftThreshold );
+    if (this->m_ApplySoftThreshold)
+      {
+      m_PhaseAnalysisFilter->SetNumOfSigmas(this->m_ThresholdNumOfSigmas);
+      }
+
+    m_FFTForwardPhaseFilter->SetInput( m_PhaseAnalysisFilter->GetOutputCosPhase() );
+
+    m_FFTForwardPhaseFilter->Update();
+    modifiedWavelets.push_back( m_FFTForwardPhaseFilter->GetOutput() );
+    modifiedWavelets.back()->DisconnectPipeline();
+    }
+
+  // ==================== Filter Set Parameters =========================
+  m_InverseWaveletFilter->SetHighPassSubBands( this->m_HighPassSubBands );
+  m_InverseWaveletFilter->SetLevels( this->m_Levels );
+  m_InverseWaveletFilter->ApplyReconstructionFactorsOff();
+  m_InverseWaveletFilter->SetInputs( modifiedWavelets );
+
+  // ==================== Filter Inter Connection =======================
+  m_InverseFFTFilter->SetInput( m_InverseWaveletFilter->GetOutput() );
+  m_ChangeInformationFilter->SetInput( m_InverseFFTFilter->GetOutput() );
```

Comment:
I don't see the options for this ChangeInformationFilter. Are you setting the reference image of this ChangeInformationFilter? It should be the output of the FFTPad I guess.